Source/WebCore/page/ElementTargetingController.cpp

```diff
@@ -101,24 +101,28 @@ static inline bool querySelectorMatchesOneElement(Document& document, const Stri
 
 struct ChildElementPosition {
     size_t index { notFound };
-    size_t childCountOfType { 0 };
+    bool firstOfType { false };
+    bool lastOfType { false };
 };
 
-static inline ChildElementPosition childIndexByType(Element& element, Element& parent)
+static inline ChildElementPosition findChild(Element& element, Element& parent)
 {
-    ChildElementPosition result;
     auto elementTagName = element.tagName();
+    RefPtr<Element> firstOfType;
+    RefPtr<Element> lastOfType;
+    size_t index = notFound;
+    size_t currentChildIndex = 0;
     for (auto& child : childrenOfType<Element>(parent)) {
-        if (child.tagName() != elementTagName)
-            continue;
-
         if (&child == &element)
-            result.index = result.childCountOfType;
-
-        result.childCountOfType++;
+            index = currentChildIndex;
```

Comment:
Optional nit: an extra line here might help with readability.
```suggestion
            index = currentChildIndex;

```